src/ddmd/objc.d

```diff
@@ -117,126 +117,140 @@ struct ObjcSelector
-                return;
+                Expression e = (*exps)[j];
+                assert(e);
+                if (e.op != TOKstructliteral)
+                    continue;
+                StructLiteralExp literal = cast(StructLiteralExp)e;
+                assert(literal.sd);
+                if (!isUdaSelector(literal.sd))
+                    continue;
+                if (fd.selector)
+                {
+                    fd.error("can only have one Objective-C selector per method");
+                    return;
+                }
+                assert(literal.elements.dim == 1);
+                StringExp se = (*literal.elements)[0].toStringExp();
+                assert(se);
+                fd.selector = ObjcSelector.lookup(cast(const(char)*)se.toUTF8(sc).string);
             }
-            assert(literal.elements.dim == 1);
-            StringExp se = (*literal.elements)[0].toStringExp();
-            assert(se);
-            fd.objc.selector = ObjcSelector.lookup(cast(const(char)*)se.toUTF8(sc).string);
         }
     }
-}
 
-extern (C++) bool objc_isUdaSelector(StructDeclaration sd)
-{
-    if (sd.ident != Id.udaSelector || !sd.parent)
-        return false;
-    Module _module = sd.parent.isModule();
-    return _module && _module.isCoreModule(Id.attribute);
-}
+    void validateSelector(FuncDeclaration fd)
+    {
+        if (!fd.selector)
+            return;
+        TypeFunction tf = cast(TypeFunction)fd.type;
+        if (fd.selector.paramCount != tf.parameters.dim)
+            fd.error("number of colons in Objective-C selector must match number of parameters");
+        if (fd.parent && fd.parent.isTemplateInstance())
+            fd.error("template cannot have an Objective-C selector attached");
+    }
 
-extern (C++) void objc_FuncDeclaration_semantic_validateSelector(FuncDeclaration fd)
-{
-    if (!fd.objc.selector)
-        return;
-    TypeFunction tf = cast(TypeFunction)fd.type;
-    if (fd.objc.selector.paramCount != tf.parameters.dim)
-        fd.error("number of colons in Objective-C selector must match number of parameters");
-    if (fd.parent && fd.parent.isTemplateInstance())
-        fd.error("template cannot have an Objective-C selector attached");
-}
+    void checkLinkage(FuncDeclaration fd)
+    {
+        if (fd.linkage != LINKobjc && fd.selector)
+            fd.error("must have Objective-C linkage to attach a selector");
+    }
 
-extern (C++) void objc_FuncDeclaration_semantic_checkLinkage(FuncDeclaration fd)
-{
-    if (fd.linkage != LINKobjc && fd.objc.selector)
-        fd.error("must have Objective-C linkage to attach a selector");
+    private bool isUdaSelector(StructDeclaration sd)
+    {
+        if (sd.ident != Id.udaSelector || !sd.parent)
+            return false;
+        Module _module = sd.parent.isModule();
+        return _module && _module.isCoreModule(Id.attribute);
+    }
 }
 
-// MARK: init
-extern (C++) void objc_tryMain_dObjc()
+private __gshared Objc objc_;
```

Comment:
What do you suggest instead, a leading `_`? The trailing underscore is there to avoid conflict with the `objc` function [1].

[1] https://github.com/dlang/dmd/pull/6679/files#diff-6c74f81a6d424346a88e4920f9f16869R245